doc/user_guide/Parameters.ipynb

```diff
@@ -784,6 +784,156 @@
     "This approach can provide significant speedup and memory savings in certain cases, but should only be used for good reasons, since it can cause confusion for any code expecting instances to be independent as they have been declared."
    ]
   },
+  {
+   "cell_type": "markdown",
+   "id": "2579c782-8dfd-4ded-995f-dc8498f4d275",
+   "metadata": {},
+   "source": [
+    "## Parameter references\n",
+    "\n",
+    "Having looked at the difference between the `Parameter` value and the `Parameter` object let us consider the idea that a `Parameter` is a reference or proxy for its underlying value. Parameters may be declared to `allow_refs` which effectively means that they will accept other Parameter **objects** (and other valid references) and resolve and reflect their current values."
```

Comment:
Why is `allow_parameters` not a more "natural" name than `allow_refs`?

Reply:
Its almost impossible for me to envision what `allow_refs` means. How should I think about it? What problem does it solve?

Reply:
Is it really a way of specifying whether the value is the value given (like a parameter) or the value the given value references (like the value of a parameter)?

Reply:
Panel has had internal support for `allow_refs` for quite a while (since 1.1.0) so you've almost certainly seen it and used it. The reason why it's not called `allow_parameters` is that it's not accurate. `refs` can be a number of things including parameters, bound functions, functions annotated with depends, reactive expressions, and even async generators. It's documented [here](https://panel.holoviz.org/how_to/interactivity/bind_component.html) and to some extent [here](https://panel.holoviz.org/how_to/interactivity/hvplot_interactive.html).

Reply:
It's also not something most users will ever have to think about as it's primarily useful for Panel developers and quite advanced Panel users (i.e. folks writing Panel apps in classes).

Reply:
>Is it really a way of specifying whether the value is the value given (like a parameter) or the value the given value references (like the value of a parameter)?

That's exactly right.